samples/nrf9160/modem_shell/src/cloud/cloud_lwm2m.c

```diff
@@ -227,6 +249,10 @@ static void cloud_lwm2m_rd_client_event_cb(struct lwm2m_ctx *client_ctx,
 	case LWM2M_RD_CLIENT_EVENT_REGISTRATION_COMPLETE:
 		mosh_print("LwM2M: Registration complete");
 		connected = true;
+		if (update_session_lifetime) {
+			/* Check if session lifetime needs to be updated. */
+			cloud_lwm2m_rd_client_update_lifetime(client.srv_obj_inst);
```

Comment:
What would happen if you did this always, i.e., remove the if-statement?
Stupid question from someone who doesn't know lwm2m at all :)

Reply:
Let's handle this in https://github.com/nrfconnect/sdk-nrf/pull/9989#discussion_r1082151367, I think these are both related to the same thing.